code/modules/events/space_ninja.dm

```diff
@@ -0,0 +1,58 @@
+/datum/round_event_control/space_ninja
+	name = "Spawn Space Ninja"
+	typepath = /datum/round_event/ghost_role/space_ninja
+	max_occurrences = 1
+	earliest_start = 20 MINUTES
+	min_players = 15
+
+/datum/round_event/ghost_role/space_ninja
+	minimum_required = 1
+	role_name = "Space Ninja"
+
+/datum/round_event/ghost_role/space_ninja/spawn_role()
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/carpspawn/C in GLOB.landmarks_list)
+		spawn_locs += (C.loc)
+	if(!spawn_locs.len)
+		message_admins("No valid spawn locations found, aborting...")
+		return MAP_ERROR
+
+	//selecting a candidate player
+	var/list/candidates = get_candidates(ROLE_NINJA, null, ROLE_NINJA)
+	if(!candidates.len)
+		return NOT_ENOUGH_PLAYERS
+
+	var/mob/dead/selected_candidate = pick(candidates)
+	var/key = selected_candidate.key
+
+	//Prepare ninja player mind
+	var/datum/mind/Mind = new /datum/mind(key)
+	Mind.assigned_role = ROLE_NINJA
+	Mind.special_role = ROLE_NINJA
+	Mind.active = TRUE
```

Comment:
Could you set these up in `mind/Initialize()` and use lower-case named variables?

Reply:
What mind/Initialize() are you referring to?  I can't anything pre-existing to utilize, do you want me to make one?

Reply:
OK, so according to mind's documentation, we shouldn't be creating minds for something like this.  So, I've made it use the default mind every mob spawns with and then assign the roles on when the space ninja antag status is gained, which also means space ninjas spawned from unconventional means now gain the proper roles as well.